x-pack/plugins/maps/public/classes/layers/layer_wizard_registry.ts

```diff
@@ -28,7 +28,9 @@ export type LayerWizard = {
   categories: LAYER_WIZARD_CATEGORY[];
   checkVisibility?: () => Promise<boolean>;
   description: string;
+  disabledReason?: string;
   icon: string | FunctionComponent<any>;
+  getIsDisabled?: () => boolean;
```

Comment:
this is a nice addition to the API. 

Not for this PR, but we should consider having the solution-layers (RUM, SIEM) use this as well. Basically, the cards would show up but show a reason like "no APM data available".

Reply:
nice idea, created https://github.com/elastic/kibana/issues/84931 to track